iroh/src/net_report.rs

```diff
@@ -443,15 +443,47 @@ impl Client {
                 let dns_resolver = self.socket_state.dns_resolver.clone();
                 let quic_client = quic_client.clone();
                 let relay_url = relay_node.url.clone();
+
+                // Regular QAD probe
                 v4_buf.spawn(
                     cancel_v4
                         .child_token()
                         .run_until_cancelled_owned(time::timeout(
                             PROBES_TIMEOUT,
-                            run_probe_v4(ip_mapped_addrs, relay_node, quic_client, dns_resolver),
+                            run_probe_v4(
+                                ip_mapped_addrs.clone(),
+                                relay_node.clone(),
+                                quic_client.clone(),
+                                dns_resolver.clone(),
+                                None,
+                            ),
                         ))
                         .instrument(info_span!("QAD-IPv4", %relay_url)),
                 );
+
+                // Port variation probe (QUIC port + 1)
+                if let Ok(relay_addr) =
+                    reportgen::get_relay_addr_ipv4(&dns_resolver, &relay_node).await
+                {
+                    let port_variation = relay_addr.port() + 1;
```

Comment:
I think assuming that this is always port + 1 is potentially problematic, I think we should make this somewhat configurable if we really need this